### DISCUSSION TITLE
为什么的这个打包的文件这么大？

### DISCUSSION THREAD

**Discussion by Lands-1203** [umijs/umi#11386] [Q&A]
![image](https://github.com/umijs/umi/assets/61341868/a6d7ad11-31f6-4fcc-9546-608aa90b05d3)

**Comment by Lands-1203** [comment]
有什么优化建议吗？

**Comment by fz6m** [comment]
这个大小是正常的，因为 antd 等第三方库都在里面。

1. 如果你启用了 `legacy` 或者设定了较低的 `target` 会使需要 polyfill 的代码变多产物体积变大，如果不需要怎么考虑兼容性，通过设定较高的 `target: { chrome: 90 }` 可以削减一些产物中的 polyfill 。

2. 参考 [代码拆分指南](https://umijs.org/blog/code-splitting) ，开启分包策略或者手动进一步拆分。

3. nginx 开启 gzip / br 等压缩，大幅减少网络传输体积。

**Reply by Lands-1203** [reply]
![image](https://github.com/umijs/umi/assets/61341868/38ec66c1-b1fa-47cf-a842-596309d03cfc)
![image](https://github.com/umijs/umi/assets/61341868/ae74bb41-38ef-408b-8777-15b326174f06)

我已经做了gzip压缩了，文件体积变小了，请求还是很长，我看了antd pro 的官网demo 400多K的文件请求时间花了5秒钟，一直900k的png才花了900ms

**Reply by Lands-1203** [reply]
我想知道这是什么原因 能科普一下吗？
![image](https://github.com/umijs/umi/assets/61341868/994f492c-5a2b-4fea-bee8-f61f957ec393)

**Reply by fz6m** [reply]
目前 antd-pro 的 preview 里不是 umi 4 的，是 umi 3 的，不过这和加载速度无关，还是要看你的网络，我的是正常的：

![screenshot_2023-07-10_21-21-50](https://github.com/umijs/umi/assets/59400654/8390966c-fdcd-4067-893c-e5ff0ae56070)

他这个站点可能部署在海外（我不了解），所以境内请求会慢一点，图片很快是因为走了境内 cdn 速度很快吧。

你在自己部署的时候要看下是不是服务器水管太小导致传输太慢，可以发到 cdn 上提速，另外你在控制台开了 disable cache 请求，正常情况 nginx 层或者 cdn 都是会对 js 、css 资源加很长周期的缓存头的（根据 hash 区分），所以用户第一次时间长点，之后就很快了。

最重要的就是网速，看直播每秒几百 kb 传输，加载个网站几百 kb js 一瞬间就下载好了，cdn 是最好的选择。
